app/celery/tasks.py

```diff
@@ -952,9 +812,19 @@ def process_incomplete_job(job_id):
     template.process_type = db_template.process_type
 
     csv = get_recipient_csv(job, template)
-    for row in csv.get_rows():
+    rows = csv.get_rows()
+    first_row = []
+    for row in rows:
         if row.index > resume_from_row:
-            process_row(row, template, job, job.service)
+            first_row.append(row)
+            for result in chunked(rows, Config.BATCH_INSERTION_CHUNK_SIZE):
+                result = first_row + result
+                process_rows(result, template, job, job.service)
+                put_batch_saving_bulk_created(
+                    metrics_logger, 1, notification_type=db_template.template_type, priority=db_template.process_type
+                )
+                first_row = []
+            exit
```

Comment:
exit?

Reply:
I think I'd prefer a `break` here instead. Makes me think we're exiting the program (which... I'm confused that we're not?)

Reply:
I changed it in the last commit